realm/realm-annotations-processor/src/test/resources/io/realm/NullTypesRealmProxy.java

```diff
@@ -1167,6 +1167,50 @@ public static NullTypes copy(Realm realm, NullTypes newObject, boolean update, M
         return realmObject;
     }
 
+    public static NullTypes createDetachedCopy(NullTypes realmObject, int currentDepth, int maxDepth, Map<RealmObject, CacheData<RealmObject>> cache) {
+        if (currentDepth > maxDepth || realmObject == null) {
```

Comment:
Integer overflow might happen if the `maxDapth == Integer.MAX_VALUE` . But not in the real life :)